### PR TITLE
watch placementscore during scheduling

### PR DESCRIPTION
**Pull request by qiujian16**
refactor eventhandler using index

Signed-off-by: Jian Qiu <jqiu@redhat.com>

### CODE REVIEW

**Comment by qiujian16** [comment]
https://github.com/open-cluster-management-io/placement/issues/87

**Comment by qiujian16** [comment]
@haoqing0110 should I add a test to ensure it can reflect on score update or there is already one?

File: pkg/controllers/scheduling/enqueue.go [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+	objs, err := e.placementIndexer.ByIndex(placementsByScore, name)
+	if err != nil {
+		runtime.HandleError(err)
+		return
+	}
+
+	for _, o := range objs {
+		placement := o.(*clusterapiv1beta1.Placement)
+		klog.V(4).Infof("enqueue placement %s/%s, because of score %s", placement.Namespace, placement.Name, key)
+		e.enqueuePlacementFunc(placement, e.queue)
```
Comment (elgnay) [inline pkg/controllers/scheduling/enqueue.go:163]: Should we consider the access scope of the placement? For example, if the score belongs to a cluster on which this placement has no permission, we can just skip this placement.

**Comment by haoqing0110** [comment]
@qiujian16 we don't have such testing so far. We have a scalability testing, but don't have the addon score in it. 
https://github.com/open-cluster-management-io/placement/tree/main/test/scalability

**Comment by haoqing0110** [comment]
@qiujian16 we have a benchmark testing on the placement decision generation with addon score. https://github.com/open-cluster-management-io/placement/blob/main/test/benchmark/benchmark_test.go. But the score doesn't change so far, may need update that.

**Comment by rokej** [comment]
I tested this with two hosting clusters. As placement score changes, I saw an immediate change in the placement decision. Thank you so much!

**Pull request merged**